src/ophyd_async/epics/odin/_odin_io.py

```diff
@@ -128,24 +146,84 @@ async def open(self, name: str, exposures_per_event: int = 1) -> dict[str, DataK
             wait_for_value(self._drv.meta_writing, "Writing", timeout=DEFAULT_TIMEOUT),
         )
 
-        return await self._describe()
+        # Add the main data
+        self._datasets = [
+            HDFDatasetDescription(
+                data_key=name,
+                dataset="/data",
```

Comment:
I would've assumed this to be `/entry/data/data`?
```suggestion
                dataset="/entry/data/data",
```